src/back/model.py

```diff
@@ -3,19 +3,20 @@
 https://huggingface.co/docs/transformers/perplexity
 """
 
-import torch
 import re
+import torch
 from transformers import GPT2LMHeadModel, GPT2TokenizerFast
 from collections import OrderedDict
 
-
 class GPT2PPL:
-    def __init__(self, model_id="gpt2"):
-        self.device = 'cuda' if torch.version.cuda is not None else 'cpu'
-        self.model_id = model_id
-        self.model = GPT2LMHeadModel.from_pretrained(model_id).to(self.device)
-        self.tokenizer = GPT2TokenizerFast.from_pretrained(model_id)
-
+    def __init__(self, device='cpu'):
+        # self.device = 'cuda' if device=='cuda' and torch.version.cuda is not None else 'cpu'

```

Comment:
this line is here because sometimes torch installs without CUDA; i'd prefer we make use of it when available (unless there's a problem with CUDA & torch besides that which warrants this)